enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+```
+
+We don't have to do anything additional to refresh the tokens, aws/eks infrastructure takes care of that for us.
+
+#### Story 8 - Procedure for hub administrator to follow, to revoke access to a managed cluster
+
+It must be possible for the hub cluster administrator to revoke access to a managed cluster. The hub admin can simply delete the `ManagedCluster` from hub and the new `aws-irsa` controller will delete all managed cluster specific resources on hub.
+
+#### Story 9 - Managed cluster permission isolation on hub
+
+One managed cluster should not be able to access resources on hub that belong to another managed cluster. 
+
+### Implementation Details/Notes/Constraints
+
+#### Changes required on hub
+The IAM roles and policies for managed cluster will be created when hub-admin accepts the registration request by running `clusteradm accept` command on hub. The role name and policies will be created using templates to ensure standardization. The role name should follow a standard naming convention so that the role on managed cluster can assume the hub role with a standardized name. It will also create an entry aws-auth configmap which will bind the iam role with permissions inside the cluster so that the incoming requests from managed cluster can have the required access inside the hub cluster.
+
+If the feature gate `ManagedClusterAutoApproval` feature gate is enabled, choosing `aws-irsa` instead of `csr` will disable [csr controller](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/registration/hub/csr/controller.go) on hub and enable a new `aws-irsa` controller, which will auto approve all registration requests, and create IAM roles and policies and RBAC permissions. 
+
+The hub registration controller must assume an IAM role with the permissions allowing it to create roles for managed clusters and attach policies. The name of this role will follow a standard template which will contain cluster name which can be parsed from the kubeconfig used to run `clusteradm init` command. The role name should follow a standard name so that the role can be created before running `clusteradm init` command and the hub controller can assume the role with a standardized name, on startup:
+```shell
+arn:aws:iam::<hub-account-id>:role/<hub-cluster-name>_managed-cluster-identity-creator
+```
+
+#### Changes required on the managed cluster
+Choosing `aws-irsa` instead of `csr` will disable [cert_controller](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/registration/clientcert/cert_controller.go) on managed cluster and enable a new `aws-irsa` controller in klusterlet/registration agent, which will create only `ManagedCluster` on hub and not CSR. The `aws-irsa` controller will watch for the updates on `ManagedCluster` , and once accepted on hub and has HubAcceptedManagedCluster condition, it will generate a kubeconfig using following command and save it to a secret called hub-kubeconfig-secret on managed cluster:
```

Comment:
Yes you are correct. If we create hub-kubeconfig without authorization on the hub, the work agent will consume the kubeconfig and [start](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/singleton/spoke/agent.go#L48) making requests to hub which will obviously fail because  authorization might not have been granted by that time, hence we decided to keep it waiting.

But please share your thoughts.